app/controllers/reports_controller.rb

```diff
@@ -12,7 +12,12 @@ def create
   end
 
   def index
-    @reports = Report.all.limit(30)
+    if params[:latitude] && params[:longitude]
+      @reports = Report
+                 .near([params[:latitude], params[:longitude]])
+    else
+      @reports = Report.all.limit(30)
+    end
```

Comment:
If they are two different things, they should have two different controller
actions.
On Dec 20, 2015 15:04, "David Elbe" <notifications@github.com> wrote:

> In app/controllers/reports_controller.rb
> <https://github.com/standout/pajat-api/pull/21#discussion_r48103681>:
>
> > @@ -12,7 +12,12 @@ def create
> >    end
> >
> >    def index
> > -    @reports = Report.all.limit(30)
> > +    if params[:latitude] && params[:longitude]
> > +      @reports = Report
> > +                 .near([params[:latitude], params[:longitude]])
> > +    else
> > +      @reports = Report.all.limit(30)
> > +    end
>
> No. listing things near a location and listing latest location is two
> different things. The first could probably use a pagination when we get a
> lot of reports, but not a limit.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/standout/pajat-api/pull/21/files#r48103681>.
>